📖 book: add documentation regarding storage drivers (#925)
kubernetes-sigs/cluster-api-provider-azure: docs/book/src/topics/external-cloud-provider.md

kube-system   cloud-node-manager-mfsdg                                            1/1     Running   0          39s
kube-system   cloud-node-manager-qrz74                                            1/1     Running   0          24s
```

## Storage Drivers

> ### Azurefile CSI Driver

Reviewer: ```suggestion
### Azure File CSI Driver
```

> Repository for Azurefile CSI Driver: https://github.com/kubernetes-sigs/azurefile-csi-driver

Reviewer: ```suggestion
Repository for Azure File CSI Driver: https://github.com/kubernetes-sigs/azurefile-csi-driver
```

> After control plane is up and running, deploy the CSI Driver:

Reviewer: ```suggestion
After the control plane is up and running, deploy the CSI Driver:
```

```bash

> kubectl --kubeconfig=./${CLUSTER_NAME}.kubeconfig \

Author: If this is correct, I'm not sure if we want to add the manifests on our side.
I prefer to point to the repo and refer the install docs from those repos.

Reviewer 2: just use below two links:
https://github.com/kubernetes-sigs/azuredisk-csi-driver/blob/master/docs/install-azuredisk-csi-driver.md
https://github.com/kubernetes-sigs/azurefile-csi-driver/blob/master/docs/install-azurefile-csi-driver.md

  apply -f templates/addons/azurefile-csi-driver.yaml
```

> After components are deployed, you should see following pods in `Running` state:

Reviewer: ```suggestion
After the Azure File CSI Driver components are deployed, you should see the following pods in a `Running` state:
```

```shell
kube-system   csi-azurefile-controller-98asdfs689-k5dab       6/6     Running   0          35s
kube-system   csi-azurefile-node-cb2s4                        3/3     Running   0          7m4s
kube-system   csi-azurefile-node-w4cvv                        3/3     Running   0          7m4s
```

> ### Azuredisk CSI Driver

Reviewer: ```suggestion
### Azure Disk CSI Driver
```

> Repository for Azuredisk CSI Driver: https://github.com/kubernetes-sigs/azuredisk-csi-driver

Reviewer: ```suggestion
Repository for Azure Disk CSI Driver: https://github.com/kubernetes-sigs/azuredisk-csi-driver
```

> After components are deployed, you should see following pods in `Running` state:

Reviewer: ```suggestion
After the Azure Disk CSI Driver components are deployed, you should see the following pods in a `Running` state:
```

```bash
kubectl --kubeconfig=./${CLUSTER_NAME}.kubeconfig \
  apply -f templates/addons/azuredisk-csi-driver.yaml
```

```shell
kube-system   csi-azuredisk-controller-23dcsss345-se3fd       6/6     Running   0          35s
kube-system   csi-azuredisk-controller-23dcsss345-6u7ys       6/6     Running   0          35s
kube-system   csi-azuredisk-node-sbsdw                        3/3     Running   0          7m4s
kube-system   csi-azuredisk-node-s34dr                        3/3     Running   0          7m4s
```

### Change default storage class

> Follow the steps bellow if you want change the current default storage class to AzureDisk CSI driver.

Reviewer: ```suggestion
Follow the steps bellow if you want change the current default storage class to Azure Disk CSI driver.
```

> First, delete the default storage class, or edit to remove the `storageclass.beta.kubernetes.io/is-default-class` annotation :

Reviewer: ```suggestion
First, delete the default storage class, or edit it to remove the `storageclass.beta.kubernetes.io/is-default-class` annotation:
```
